### PR TITLE
Add BBE for HTTP secure client

### PR DESCRIPTION
**Pull request by ldclakmal**
## Purpose
Fixes https://github.com/ballerina-platform/ballerina-examples/issues/951

### CODE REVIEW

File: examples/secured-client-with-basic-auth/secured_client_with_basic_auth.bal [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import ballerina/http;
+import ballerina/log;
+import ballerina/config;
```
[the remark this answers is not in the document]
Comment (ldclakmal) [reply examples/secured-client-with-basic-auth/secured_client_with_basic_auth.bal]: Fixed.

File: examples/secured-client-with-basic-auth/secured_client_with_basic_auth.bal [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+};
+
+function main(string... args) {
+    // This defines the authentication credentials of the HTTP service.
+    config:setConfig("b7a.users.tom.password", "1234");
+
+    // Send a `GET` request to the specified endpoint.
+    var response = httpEndpoint->get("/hello/sayHello");
+    match response {
+        http:Response resp => log:printInfo(check resp.getPayloadAsString());
```
[the remark this answers is not in the document]
Comment (ldclakmal) [reply examples/secured-client-with-basic-auth/secured_client_with_basic_auth.bal]: Fixed.

File: examples/secured-client-with-basic-auth/secured_client_with_basic_auth.bal [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    authConfig: {
+        authentication: { enabled: true }
+    }
+}
+service<http:Service> echo bind ep {
+    @http:ResourceConfig {
+        methods: ["GET"],
+        path: "/sayHello"
+    }
+
```
[the remark this answers is not in the document]
Comment (ldclakmal) [reply examples/secured-client-with-basic-auth/secured_client_with_basic_auth.bal]: Fixed.

File: examples/secured-client-with-basic-auth/secured_client_with_basic_auth.description [hunk 4]
```diff
@@ -0,0 +1,4 @@
+// A secured client with basic authentication is used to connect to a
+// secured service with basic authentication.
+// Basic authentication is enabled by setting the `scheme: http:BASIC_AUTH`
+// The `username` and `password` should be specified as needed.
```
[the remark this answers is not in the document]
Comment (ldclakmal) [reply examples/secured-client-with-basic-auth/secured_client_with_basic_auth.description]: Fixed.

File: examples/secured-client-with-basic-auth/secured_client_with_basic_auth.out [hunk 5]
```diff
@@ -0,0 +1,5 @@
+# Create a file named `secured_client_with_basic_auth.bal`, copy the code
+# into it, and use the `ballerina run` command to start the service.
+$ ballerina run secured_client_with_basic_auth.bal
+ballerina: started HTTPS/WSS endpoint 0.0.0.0:9090
+2018-07-31 18:00:07,931 INFO  [] - Hello, World!!!
```
[the remark this answers is not in the document]
Comment (ldclakmal) [reply examples/secured-client-with-basic-auth/secured_client_with_basic_auth.out]: Fixed.

File: examples/secured-client-with-basic-auth/secured_client_with_basic_auth.bal [hunk 6]
```diff
@@ -0,0 +1,61 @@
+import ballerina/config;
+import ballerina/http;
+import ballerina/log;
+
+// Define the basic auth client endpoint to call the backend services.
+// Basic authentication is enabled by setting the `scheme: http:BASIC_AUTH`
```
Comment (chamil321) [inline examples/secured-client-with-basic-auth/secured_client_with_basic_auth.bal]: This is description is quite same with secured_client_with_basic_auth.description. Shall we refactor?

File: examples/secured-client-with-basic-auth/secured_client_with_basic_auth.out (hunk 5 above)
Comment (Bhashinee) [inline examples/secured-client-with-basic-auth/secured_client_with_basic_auth.out]: Shall we remove the date here? 
"INFO  [] - Hello, World!!!" will be enough.

File: examples/secured-client-with-basic-auth/secured_client_with_basic_auth.bal (hunk 6 above)
Comment (ldclakmal) [reply examples/secured-client-with-basic-auth/secured_client_with_basic_auth.bal]: Fixed

File: examples/secured-client-with-basic-auth/secured_client_with_basic_auth.out (hunk 5 above)
Comment (ldclakmal) [reply examples/secured-client-with-basic-auth/secured_client_with_basic_auth.out]: Fixed.

**Review by chamil321 — APPROVED** [review]

File: examples/secured-client-with-basic-auth/secured_client_with_basic_auth.description [hunk 7]
```diff
@@ -0,0 +1,6 @@
+// A secured client with basic authentication is used to connect to a
+// secured service with basic authentication. An additional `auth`
+// field is added to the HTTP client endpoint initialization in order
+// to make the simple HTTP client endpoint into a secure client endpoint.
+// Inside the `auth` filed the `scheme`, `username` and `password`
```
Comment (samgnaniah) [inline examples/secured-client-with-basic-auth/secured_client_with_basic_auth.description]: Typo. Change 'filed' to 'field'.

File: examples/secured-client-with-jwt-auth/secured_client_with_jwt_auth.description [hunk 8]
```diff
@@ -0,0 +1,5 @@
+// A secured client with jwt authentication is used to connect to a
+// secured service with jwt authentication. An additional `auth`
+// field is added to the HTTP client endpoint initialization in order
+// to make the simple HTTP client endpoint into a secure client endpoint.
+// Inside the `auth` filed the `scheme` is defined for JWT authentication.
```
Comment (samgnaniah) [inline examples/secured-client-with-jwt-auth/secured_client_with_jwt_auth.description]: Same typo as before. Fix in other similar places.

File: examples/secured-client-with-jwt-auth/secured_client_with_jwt_auth.description [hunk 9]
```diff
@@ -0,0 +1,5 @@
+// A secured client with jwt authentication is used to connect to a
```
Comment (samgnaniah) [inline examples/secured-client-with-jwt-auth/secured_client_with_jwt_auth.description]: Shall we change 'jet' to 'JWT'? Fix everywhere to maintain consistency.

File: examples/secured-client-with-oauth2/secured_client_with_oauth2.description [hunk 10]
```diff
@@ -0,0 +1,7 @@
+// A secured client with oauth2 authentication is used to connect to a
```
Comment (samgnaniah) [inline examples/secured-client-with-oauth2/secured_client_with_oauth2.description]: Change 'oauth2' to 'OAuth2'. Let's fix everywhere for consistency.

File: examples/secured-client-with-basic-auth/secured_client_with_basic_auth.description (hunk 7 above)
Comment (ldclakmal) [reply examples/secured-client-with-basic-auth/secured_client_with_basic_auth.description]: Fixed.

File: examples/secured-client-with-jwt-auth/secured_client_with_jwt_auth.description (hunk 8 above)
Comment (ldclakmal) [reply examples/secured-client-with-jwt-auth/secured_client_with_jwt_auth.description]: Fixed.

File: examples/secured-client-with-jwt-auth/secured_client_with_jwt_auth.description (hunk 9 above)
Comment (ldclakmal) [reply examples/secured-client-with-jwt-auth/secured_client_with_jwt_auth.description]: Fixed.

File: examples/secured-client-with-oauth2/secured_client_with_oauth2.description (hunk 10 above)
Comment (ldclakmal) [reply examples/secured-client-with-oauth2/secured_client_with_oauth2.description]: Fixed.

**Pull request merged**